h/services/delete_user.py

```diff
@@ -23,17 +23,36 @@ def delete(self, user):
         message.
         """
 
-        if Group.created_by(self.request.db, user).count() > 0:
-            raise UserDeleteError('Cannot delete user who is a group creator.')
-
-        user.groups = []
+        # Check whether this user has created any groups that others have
+        # annotated in.
+        #
+        # We check for non-empty `group_ids` before querying the DB to avoid an
+        # expensive SQL query if `in_` is given an empty list (see
+        # https://stackoverflow.com/questions/23523147/)
+        created_groups = self.request.db.query(Group) \
+                                        .filter(Group.creator == user)
+        group_ids = [g.pubid for g in created_groups]
+        if len(group_ids) > 0:
+            other_user_group_anns = self.request.db.query(Annotation) \
```

Comment:
A fairly minor point: the name of this variable seems to suggest it contains the annotations themselves, rather than a count of how many there are.

Reply:
Agreed. I've changed this.